source/server/config/network/client_ssl_auth.cc

```diff
@@ -1,34 +1,50 @@
+#include "client_ssl_auth.h"
+
 #include "envoy/network/connection.h"
 #include "envoy/server/instance.h"
 
 #include "common/filter/auth/client_ssl.h"
-#include "server/configuration_impl.h"
 
 namespace Server {
 namespace Configuration {
 
-/**
- * Config registration for the client SSL auth filter. @see NetworkFilterConfigFactory.
- */
-class ClientSslAuthConfigFactory : public NetworkFilterConfigFactory {
-public:
-  // NetworkFilterConfigFactory
-  NetworkFilterFactoryCb tryCreateFilterFactory(NetworkFilterType type, const std::string& name,
-                                                const Json::Object& json_config,
-                                                Server::Instance& server) {
-    if (type != NetworkFilterType::Read || name != "client_ssl_auth") {
-      return nullptr;
-    }
-
-    Filter::Auth::ClientSsl::ConfigPtr config(Filter::Auth::ClientSsl::Config::create(
-        json_config, server.threadLocal(), server.clusterManager(), server.dispatcher(),
-        server.stats(), server.random()));
-    return [config](Network::FilterManager& filter_manager) -> void {
-      filter_manager.addReadFilter(
-          Network::ReadFilterPtr{new Filter::Auth::ClientSsl::Instance(config)});
-    };
+const std::string ClientSslAuthConfigFactory::CLIENT_SSL_SCHEMA(R"EOF(
+  {
+    "$schema": "http://json-schema.org/schema#",
+    "properties": {
+      "auth_api_cluster" : {"type" : "string"},
+      "stat_prefix" : {"type" : "string"},
+      "ip_white_list" : {
+        "type" : "array",
+        "items" : {
+          "type": "string",
+          "format" : "ipv4"
+        }
+      }
+    },
+    "required": ["auth_api_cluster", "stat_prefix"],
+    "additionalProperties": false
   }
-};
+  )EOF");
+
+NetworkFilterFactoryCb
+ClientSslAuthConfigFactory::tryCreateFilterFactory(NetworkFilterType type, const std::string& name,
+                                                   const Json::Object& json_config,
+                                                   Server::Instance& server) {
+  if (type != NetworkFilterType::Read || name != "client_ssl_auth") {
+    return nullptr;
+  }
+
+  json_config.validateSchema(CLIENT_SSL_SCHEMA);
+
+  Filter::Auth::ClientSsl::ConfigPtr config(Filter::Auth::ClientSsl::Config::create(
```

Comment:
The schema validation should be done closer to the relevant config code, so that unit tests are covered also. For example, in this case, it should be done inside Filter::Auth::ClientSsl::Config constructor. (Same applies for all the other cases). As long as you already did the h/cc splits for the filters, please keep that, and make sure all of the config stuff has code coverage, since we were missing that previously.